### DISCUSSION TITLE
StatusBarBehavior No Longer Working on iOS With Exception

### DISCUSSION THREAD

**Discussion by Nope-Git** [CommunityToolkit/Maui#2280] [Q&A] [answered]
I'm having trouble with StatusBarBahavior on MAUI iOS. I've tried with iOS 17.5 and 18. My StatusBarBehavior code worked fine before updating MAUI, the MAUI tookit, and XCode. Now I get the following error message. 

**Microsoft.Maui.Controls.Xaml.XamlParseException:** 'Position 8:10. Type toolkit:StatusBarBehavior not found in xmlns http://schemas.microsoft.com/dotnet/2022/maui/toolkit'

I have the following code in my XAML.
```
    <ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui" 
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml" 
             x:Class="MyApp.Pages.CoreAppPage" 
             Title="MyApp" 
             xmlns:ViewModels="clr-namespace:MyApp.ViewModels" 
             xmlns:toolkit="http://schemas.microsoft.com/dotnet/2022/maui/toolkit"> 
    <Page.Behaviors> 
        <toolkit:StatusBarBehavior 
            StatusBarColor="{StaticResource StatusBarColor}"
            StatusBarStyle="LightContent" />
    </Page.Behaviors>... 
````
MauiProgram.cs
```
    var builder = MauiApp.CreateBuilder();
            builder
                .UseMauiApp<App>()
                // Initialize the .NET MAUI Community Toolkit by adding the below line of code
                .UseMauiCommunityToolkit()
                .ConfigureFonts(fonts =>
                {
                    fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
                    fonts.AddFont("OpenSans-Semibold.ttf", "OpenSansSemibold");
                });
```

I've tried restarting both the mac I'm building on and the machine that runs Visual Studio. The bin and object folders were deleted and rebuilt but produced the same error. As stated previously, this was something that was working fine before I updated.

Is this a bug or are there some other things I should be trying to get the app to build after the update?

**Comment by Nope-Git** [comment]
Downgrading the community toolkit to 9.0.3 appears to have fixed this behavior.

**Comment by raver99** [comment, marked as answer]
I think it has to do with the linking step. If I disabled the Linker, it worked. Then I created an instance of the StatusBarBehavior in code to help the linker pick it up and it worked.
